src/api/users/users.schema.ts

```diff
@@ -0,0 +1,13 @@
+export interface CreateUserRequestBody {
+  email: string;
+  password: string;
+  name: string;
+  birthdate: string;
+}
+
+export interface CreateUserResponse {
+  id: number;
+  name: string;
+  email: string;
+  birthdate: Date;
+}
```

Comment:
```suggestion
export interface CreateUserRequestBody extends UserInput {}

export interface CreateUserResponse extends User {}
```

Esses são 2 "pulos do gato" que usamos na arquitetura, pra simplificar. Nós costumamos fazer com que os models de apresentação (camada de api) sejam implementações dos models de domain, assim não precisamos ficar mapeando um pro outro necessariamente.

Claro que às vezes existem diferenças entre eles e nesse caso precisamos mapear, mas eu diria que 90% das vezes dá pra seguirmos com o mesmo modelo.